subprojects/core/src/main/java/org/gradle/api/internal/AbstractTask.java

```diff
@@ -1069,6 +1094,17 @@ public List<ResourceLock> getSharedResources() {
         return getBuildServiceRegistry().getSharedResources(taskRequiredServices.getElements());
     }
 
+    private void addProviderTaskDependency(Provider<?> provider) {
+        // See DefaultTaskDependency.visitDependencies
+        if (provider instanceof ProviderInternal) {
+            ProviderInternal<?> providerInternal = (ProviderInternal<?>) provider;
+            ValueSupplier.ValueProducer producer = providerInternal.getProducer();
+            if (producer.isKnown()) {
```

Comment:
❌ This is too early for this check. It should work to unconditionally add the provider to the `dependencies` collection.

Reply:
See https://github.com/gradle/gradle/pull/24612